File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/application/user/command/ReturnBookService.java
package dev.starryeye.book_library.application.user.command;

import dev.starryeye.book_library.domain.user.User;
import dev.starryeye.book_library.domain.user.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Transactional
@Service
@RequiredArgsConstructor
public class ReturnBookService {

    private final UserRepository userRepository;

    public void returnBook(Long userId, Long bookId) {

        User user = userRepository.findByIdWithLoanHistories(userId)
                .orElseThrow(() -> new IllegalArgumentException("user is not found, id = " + userId));

        user.returnBook(bookId);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/user/command/ReturnBookService.kt
package dev.starryeye.book_library.application.user.command

import dev.starryeye.book_library.domain.user.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional
@Service
class ReturnBookService(
    private val userRepository: UserRepository,
) {

    fun returnBook(userId: Long, bookId: Long) {

        val user = userRepository.findByIdWithLoanHistories(userId)
            ?: throw IllegalArgumentException("user is not found, id = $userId")

        user.returnBook(bookId)
    }
}